src/Microsoft.Data.Sqlite.Core/SqliteDataRecord.cs

```diff
@@ -53,7 +54,7 @@ protected override int GetSqliteType(int ordinal)
         protected override T GetNull<T>()
             => typeof(T) == typeof(DBNull) || typeof(T) == typeof(object)
                 ? (T)(object)DBNull.Value
-                : throw new InvalidCastException();
+                : throw new SqlNullValueException();
```

Comment:
I think this type is SQL Client specific and we should introduce our own type. I'll look more into it...

Reply:
Confirmed.

> The System.Data.SqlTypes namespace provides classes for native data types in SQL Server.

I think we should just throw `InvlaidOperationException` with a message like this:

> The data is NULL. This method can't be called on NULL values. Check using IsDBNull before calling.